### PR TITLE
re/nre: mention nim-regex as the alternative

### PR DESCRIPTION
**Pull request by Araq**

### CODE REVIEW

**Comment by juancarlospaco** [comment]
I agree; But this kinda leaves stdlib without Regex, thats a big missing feature compared to other programming languages...

Work to integrate `regex` into stdlib should be considered soon.

**Comment by dom96** [comment]
Agreed that we should incorporate this awesome library into stdlib.

There are multiple ways we can achieve this, depends on how the library's author wants the library to evolve (outside stdlib in its own repo or not). If in its own repo then the only way I can see this working is if we implement a feature in Nimble that allows packages to be included as part of Nim, and that then Nimble is aware that these packages are "installed".

**Comment by Araq** [comment]
> If in its own repo then the only way I can see this working is if we implement a feature in Nimble that allows packages to be included as part of Nim, and that then Nimble is aware that these packages are "installed".

The "nim distro" idea. It's time to do it.

**Comment by konsumlamm** [comment]
Why not use one of the adminitions (like `.. note::` or `.. warning::`)?

**Comment by timotheecour** [comment]
pros of pkg/regex: better API, pure nim
works at CT is potentially a moot advantage because of https://github.com/nitely/nim-regex/issues/104 and because a vmops might work for re/nre (and maybe pkg/regex too)

performance wise, see this thread: https://github.com/nitely/nim-regex/issues/57 tracking performance which was initially showing regex being 40x slower but @nitely has improved performance a lot since then, eg see https://github.com/nitely/nim-regex/pull/58#issuecomment-659683459.

IMO we should still provide a comprehensive (and easy to re-run / update) benchmark on a variety of tasks before declaring pkg/regex as "the one true choice"

**Comment by nitely** [comment]
All I can say is nim-regex depends on `unicodedb`, and I think that makes it a no-go for including it into the stdlib.

@timotheecour performance concerns are valid, we need more benchmarks.

Why is re/nre outdated anyway? is this about the APIs, requiring pcre VS no dependencies, or what exactly?

**Comment by Araq** [comment]
For me it's about "requiring pcre VS no dependencies" plus the fact that both re/nre are hardly maintained.

**Comment by Varriount** [comment]
> I agree; But this kinda leaves stdlib without Regex, thats a big missing feature compared to other programming languages...
> 
> Work to integrate `regex` into stdlib should be considered soon.

Or, we can just recommend people download the package, while carefully considering whether that implementation is suitable for inclusion into the standard library in the first place. Nim has already gone through 2 regex modules, let's try to avoid continuing the trend.

File: lib/impure/re.nim [hunk 1]
```diff
@@ -10,6 +10,9 @@
 when defined(js):
   {.error: "This library needs to be compiled with a c-like backend, and depends on PCRE; See jsre for JS backend.".}
 
+## NOTE: The `re` module is outdated, but going to be kept for backward
```
Comment (ringabout) [inline lib/impure/re.nim:13]:
```suggestion
## .. note:: The `re` module is outdated, but going to be kept for backward
```
ditto

File: lib/impure/nre.nim [hunk 2]
```diff
@@ -9,6 +9,9 @@
 when defined(js):
   {.error: "This library needs to be compiled with a c-like backend, and depends on PCRE; See jsre for JS backend.".}
 
+## NOTE: The `nre` module is outdated, but going to be kept for backward
```
Comment (ringabout) [inline lib/impure/nre.nim:12]:
```suggestion
## .. note:: The `nre` module is outdated, but going to be kept for backward
```
note or warning

**Comment by juancarlospaco** [comment]
Maybe a tiny new module on top of existing Regexes, multi-backend regex, kinda like: 

```nim
initRe(backend = JsRe)   # backend = Nre , backend = X
let r: Regex = /[0-9]/i  # Generates newRegex("[0-9]", "i")
```

Then maybe users can "hook up" their own:

```nim
initRe(backend = CustomTypeOrConcept)  # Backend delegated to user.
let r: Regex = /[0-9]/i  # Features and Bugs here delegated to user.
```

**Pull request closed without merging**